File: tests/go/cases/case_008_publish_inventory.go
package escape_tests

var retainedCase008 = []map[string]string{}

func Case008PublishInventory(input string) string {
    // Task case 008: publish inventory payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "publish_inventory",
        "entity": "inventory",
        "stage": "publish",
        "input": raw,
    }
    // ESCAPE: payload is captured by retained closure metadata.
    signature := payload["task"] + ":" + payload["input"]
    retainedCase008 = append(retainedCase008, map[string]string{"handler": signature})
    return "ok"
}

File: tests/python/cases/case_008_publish_inventory.py
"""
Task case 008: publish inventory payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_008_publish_inventory(input_data):
    # Task: publish inventory records and prepare transport-ready payload.
    task_name = 'publish_inventory'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'inventory',
        'stage': 'publish',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: closure captures payload and handler is retained in RETAINED_HANDLERS.
    def handler():
        return payload['checksum']
    RETAINED_HANDLERS.append(handler)
    return 'ok'
